airflow/providers/databricks/hooks/databricks_base.py

```diff
@@ -374,6 +377,95 @@ async def _a_get_aad_token(self, resource: str) -> str:
 
         return jsn["access_token"]
 
+    def _get_aad_token_for_workload_identity(self, resource: str) -> str:
```

Comment:
In principle getting a token in this manner isn't restricted to workload identity perse. For example, using the `DefaultAzureCredential` you can support any of the authentication options listed here: https://learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/credential-chains?tabs=dac#defaultazurecredential-overview. 

Would it make sense to rename this method to reflect this and maybe use an options such as `use_azure_default_credential` instead?

Reply:
Great suggestion. The change has been made!